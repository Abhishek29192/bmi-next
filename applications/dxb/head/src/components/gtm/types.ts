export type DataLayer = Record<string, unknown>[];

declare global {
  interface Window {
    dataLayer: DataLayer;
    [key: string]: DataLayer;
  }
}

export type GTMParams = {
  gtmIds: string[];
  dataLayer?: string[];
  dataLayerName?: string;
  auth?: string;
  preview?: string;
  includeInDevelopment?: boolean;
  enableWebVitalsTracking?: boolean;
  selfHostedOrigin?: string;
  defaultDataLayer?: { platform: string; env: string };
};

export type GAParams = {
  gaId: string;
  dataLayerName?: string;
};
