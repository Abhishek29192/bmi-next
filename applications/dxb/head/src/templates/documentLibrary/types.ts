import { Data as BreadcrumbsData } from "../../components/Breadcrumbs";
import { Data as PageData } from "../../components/Page";
import { Data as PageInfoData } from "../../components/PageInfo";
import { RichTextData } from "../../components/RichText";
import { Data as SiteData } from "../../components/Site";
import { DocumentsFilters } from "../../schema/resolvers/types/DocumentsFilters";
import { ContentfulDocument } from "../../types/Document";
import { ProductDocument } from "../../types/pim";
import { ResultType, Source, URLFilter } from "../../utils/filters";

export type QueryParams = {
  filters: URLFilter[];
};

export interface DocumentLibraryPageContext {
  pageId: string;
  siteId: string;
  categoryCode: string;
  variantCodeToPathMap?: Record<string, string>;
}

export interface ContentfulDocumentLibraryPage extends PageInfoData, PageData {
  description: RichTextData | null;
  allowFilterBy: string[] | null;
  source: Source;
  resultsType: ResultType;
  breadcrumbs: BreadcrumbsData;
  categoryCodes: string[];
  breadcrumbTitle: string;
  documentsFilters: DocumentsFilters | null;
}

export interface DocumentLibraryProps {
  pageContext: DocumentLibraryPageContext;
  data: {
    contentfulDocumentLibraryPage: ContentfulDocumentLibraryPage;
    contentfulSite: SiteData;
  };
}

export type DocumentType =
  | ContentfulDocument["__typename"]
  | ProductDocument["__typename"]
  | "";
