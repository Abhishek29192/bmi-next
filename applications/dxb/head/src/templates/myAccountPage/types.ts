import { Data as ContactDetailsData } from "../../components/ContactDetails";
import type { Data as BreadcrumbsData } from "../../components/Breadcrumbs";
import type { Data as SiteData } from "../../components/Site";
import type { HeroProps } from "./Hero";
import type { Data as LinkData } from "../../components/link/types";

export type ServiceSupportSectionProps = {
  titleForServiceSupportSection: string;
  serviceSupportCards: [ContactDetailsData, ...ContactDetailsData[]];
};

export type GlobalTools = "My profile" | "Trainings" | "Roof measurement";

export type LocalToolType =
  | "Calculator"
  | "Company Profile"
  | "Guarantees"
  | "Team"
  | "Visualiser";

export type LocalTool = {
  title: string;
  type: LocalToolType;
  link: LinkData;
};

export type ToolSectionProps = {
  titleForToolSection: string;
  globalTools: readonly [GlobalTools, ...GlobalTools[]];
  tools: readonly [LocalTool, ...LocalTool[]] | null;
  path: string;
};

export type AccountPage = {
  title: string;
  breadcrumbTitle: string | null;
  breadcrumbs: BreadcrumbsData;
} & ToolSectionProps &
  Omit<ServiceSupportSectionProps, "serviceSupportCards"> &
  HeroProps & {
    path: string;
    serviceSupportCards:
      | ServiceSupportSectionProps["serviceSupportCards"]
      | null;
  };

export type SiteDataWithAccountPage = Omit<SiteData, "accountPage"> & {
  accountPage: AccountPage;
};
