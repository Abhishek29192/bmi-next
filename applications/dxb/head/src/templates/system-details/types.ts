export type GalleryImageType = {
  mainSource: string;
  thumbnail: string;
  altText: string;
};

export type SystemProductImageType = SystemImageType & {
  format: string;
  containerId: string;
};

export type SystemImageType = {
  url: string;
  name: string;
  mime: string;
  fileSize: number;
  containerId: string;
  assetType: string;
  altText?: string;
  allowedToDownload: boolean;
  realFileName: string;
};

export type Category = {
  categoryType: string;
  name: string;
  image?: SystemImageType;
};

export type Feature = {
  code:
    | "bmiSystemsClassificationCatalog/1.0/systemAttributes.roofbuildup"
    | "bmiSystemsClassificationCatalog/1.0/systemAttributes.promotionalcontent";
  featureValues: { value: string }[];
  name: string;
};

export type Classification = {
  code: "systemAttributes" | "measurementAttributes";
  features: Feature[];
  name: string;
};

export interface SystemDetails {
  type?: string;
  approvalStatus?: string;
  code?: string;
  assets: any[];
  name: string;
  categories: Category[];
  classifications: Classification[];
  images: SystemProductImageType[] | null;
  longDescription?: string;
  shortDescription?: string;
  systemBenefits?: string[];
  systemLayers?: any[];
  systemReferences?: any[];
}
