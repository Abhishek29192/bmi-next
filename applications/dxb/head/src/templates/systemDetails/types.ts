import {
  Image,
  ClassificationFeatureUnit,
  Product
} from "../../components/types/ProductBaseTypes";

export type GalleryImageType = {
  mainSource: string;
  thumbnail: string;
  altText: string;
};

export type CategoryImage = {
  allowedToDownload: boolean;
  fileSize: number;
  mime: string;
  name: string;
  realFileName: string;
  url: string;
};

export type Category = {
  categoryType: string;
  name: string;
  image?: CategoryImage;
  code?: string;
};

export type FeatureValue = { value: string };

export type Feature = {
  code: string;
  featureValues: FeatureValue[];
  name: string;
  featureUnit?: ClassificationFeatureUnit;
};

export type Classification = {
  code:
    | "systemAttributes"
    | "measurementAttributes"
    | "scoringWeightAttributes"
    | "ScoringWeightAttributes"
    | "uniquesellingpropositionsAttributes";
  features: Feature[];
  name: string;
};

export type AssetType =
  | "BIM"
  | "WARRANTIES"
  | "AWARDS"
  | "GUARANTIES"
  | "CERTIFICATES"
  | "SPECIFICATION"
  | "CAD"
  | "DATA_SHEETS";

export type MimeType =
  | "image/jpeg"
  | "image/png"
  | "image/tiff"
  | "application/pdf";

export type Assets = {
  allowedToDownload: boolean;
  assetType: AssetType;
  fileSize?: number;
  mime: MimeType;
  name: string;
  realFileName: string; // includes file extension
  url: string;
  format?: string;
};

export type SystemBenefits = string[];

interface ProductCode {
  code: string;
}

export type SystemLayer = {
  addon: boolean;
  approvalStatus: "approved" | "unapproved";
  code: string;
  images: Image[];
  layerNumber: number;
  longDescription: string;
  name: string;
  optionalProducts: ProductCode[];
  products: ProductCode[];
  shortDescription: string;
  type: string;
  relatedOptionalProducts: Product[];
  relatedProducts: Product[];
};

export interface SystemDetails {
  type?: string;
  approvalStatus?: string;
  code?: string;
  assets: Assets[];
  name: string;
  categories: Category[];
  classifications: Classification[];
  images: Image[] | null;
  longDescription: string;
  shortDescription?: string;
  systemBenefits: SystemBenefits;
  systemLayers?: SystemLayer[];
  systemReferences?: any[];
}
