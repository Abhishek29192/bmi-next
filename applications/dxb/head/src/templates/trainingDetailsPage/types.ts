import type { Course } from "@bmi/gatsby-source-docebo";
import type { Data as BreadcrumbsData } from "../../components/Breadcrumbs";

export type Session = {
  id: number;
  code: string | null;
  name: string;
  date_start: number;
  date_end: number;
};

export type TrainingDetailsCourseType = Pick<
  Course,
  | "code"
  | "id_course"
  | "slug_name"
  | "name"
  | "course_type"
  | "categoryName"
  | "description"
  | "img_url"
  | "price"
  | "currencySymbol"
> & {
  breadcrumbs: BreadcrumbsData;
  sessions: Session[];
};
