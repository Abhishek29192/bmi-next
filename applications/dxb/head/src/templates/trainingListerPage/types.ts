import { Training } from "@bmi/elasticsearch-types";
import { Data as PageData } from "../../components/Page";
import { Data as SiteData } from "../../components/Site";
import { Data as ImageData } from "../../components/Image";
import { Data as TitleWithContentData } from "../../components/TitleWithContent";

export type EsCollapsedTraining = {
  inner_hits: {
    inner_hits: {
      hits: {
        total: {
          value: number;
          relation: string;
        };
        max_score: number;
        hits: {
          _index: string;
          _type: "_doc";
          _id: string;
          _score: number;
          _source: Training;
        }[];
      };
    };
  };
  _source: Training;
};

export type EsTrainingHit = {
  _index: string;
  _type: "_doc";
  _id: string;
  _score: number;
  _source: Training;
};

export type EsResponse<T extends EsCollapsedTraining | EsTrainingHit> = {
  hits: {
    total: {
      value: number;
      relation: string;
    };
    max_score: number | null;
    hits: T[];
  };
};

export type TrainingListerPageData = Omit<PageData, "signupBlock"> & {
  __typename: "ContentfulTrainingListerPage";
  title: string;
  subtitle: string | null;
  breadcrumbTitle: string | null;
  featuredMedia: ImageData;
  searchTips: TitleWithContentData;
};

export type TrainingListerPageProps = {
  data: {
    contentfulTrainingListerPage: TrainingListerPageData;
    contentfulSite: SiteData;
  };
  pageContext: {
    variantCodeToPathMap?: Record<string, string>;
  };
};
