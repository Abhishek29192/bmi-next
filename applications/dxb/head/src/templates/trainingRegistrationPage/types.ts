import type { Training } from "@bmi/elasticsearch-types";
import type { ESResponse } from "../../types/elasticsearch";
import type { Data as SiteData } from "../../components/Site";
import type { Data as BreadcrumbsData } from "../../components/Breadcrumbs";

export type TrainingRegistrationPageData = {
  __typename: "ContentfulTrainingRegistrationPage";
  breadcrumbs: BreadcrumbsData;
  breadcrumbTitle: string | null;
  path: string;
  title: string;
  salutationTitle: string;
  salutationMale: string;
  salutationFemale: string;
  firstName: string;
  lastName: string;
  email: string;
  companyName: string;
  position: string;
  customerNumber: string;
  city: string;
  street: string;
  postalCode: string;
  phoneNumber: string;
  competentChamberLabel: {
    competentChamberLabel: string;
  };
  competentChamber: string;
  bmiSystemPartnerClubTitle: string;
  isMemberOfBmiLabel: string;
  isNotMemberOfBmiLabel: string;
  discoverySourceTitle: string;
  discoverySourceBrochure: string;
  discoverySourceFieldService: string;
  discoverySourceWebsite: string;
  discoverySourceFacebook: string;
  discoverySourceInstagram: string;
  discoverySourceXing: string;
  discoverySourceLinkedin: string;
  discoverySourceOther: string;
  discoverySourceSpecifyOther: string;
  comment: string;
  consentText: {
    consentText: string;
  };
  termsOfUse: {
    termsOfUse: string;
  };
  registerButton: string;
};

export type TrainingRegistrationPageProps = {
  data: {
    contentfulTrainingRegistrationPage: TrainingRegistrationPageData;
    contentfulTrainingListerPage: {
      path: string;
      breadcrumbs: BreadcrumbsData;
    } | null;
    contentfulSite: SiteData;
  };
};

export type ESTrainingDetails = ESResponse<Training>;
