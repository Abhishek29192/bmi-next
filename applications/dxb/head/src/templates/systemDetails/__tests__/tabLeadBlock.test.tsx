import React from "react";
import { render, cleanup } from "@testing-library/react";
import Component from "../tabLeadBlock";
import dataJson from "../../../data/pim-mock-data.json";
import "@testing-library/jest-dom";
import { Assets } from "../types";
import { renderWithRouter } from "../../../test/renderWithRouter";

const guaranteesAndWarranties = dataJson.assets.filter(
  ({ assetType }) => assetType === "WARRANTIES"
) as Assets[];

const awardsAndCertificates = dataJson.assets.filter(
  ({ assetType }) => assetType === "AWARDS"
) as Assets[];

describe("TabLeadBlock tests", () => {
  afterEach(cleanup);

  it("should render", () => {
    const { container, getByText } = render(
      <Component
        longDescription={dataJson.longDescription}
        guaranteesAndWarranties={guaranteesAndWarranties}
        awardsAndCertificates={awardsAndCertificates}
      />
    );

    const aboutTabButton = getByText("sdp.leadBlock.about", {
      exact: false
    });

    expect(container).toMatchSnapshot();
    expect(aboutTabButton).toBeInTheDocument();
  });

  it("should render the bimIframe tab", () => {
    const { container } = renderWithRouter(
      <Component
        longDescription={dataJson.longDescription}
        guaranteesAndWarranties={guaranteesAndWarranties}
        awardsAndCertificates={awardsAndCertificates}
        bimIframeUrl="https://google.com"
      />
    );
    expect(container.querySelector("#tabpanel-four")).toMatchSnapshot();
  });

  it("should not render the bimIframe tab", () => {
    const { container } = renderWithRouter(
      <Component
        longDescription={dataJson.longDescription}
        guaranteesAndWarranties={guaranteesAndWarranties}
        awardsAndCertificates={awardsAndCertificates}
        bimIframeUrl={null}
      />
    );
    expect(container.querySelector("#tabpanel-four")).toBe(null);
  });
});
