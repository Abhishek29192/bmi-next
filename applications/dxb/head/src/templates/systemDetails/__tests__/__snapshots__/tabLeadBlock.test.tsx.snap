// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabLeadBlock tests should render 1`] = `
<div>
  <div
    class="Section Section--white"
  >
    <div
      class="Container"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl"
      >
        <div
          class="Tabs Tabs--primary"
        >
          <div
            class="TabsBar"
          >
            <div
              class="Container"
            >
              <div
                class="MuiContainer-root MuiContainer-maxWidthXl"
              >
                <div
                  class="MuiTabs-root"
                >
                  <div
                    class="MuiTabs-scrollable"
                    style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
                  />
                  <div
                    class="MuiTabs-scroller MuiTabs-scrollable"
                    style="margin-bottom: 0px;"
                  >
                    <div
                      aria-label="tabs"
                      class="MuiTabs-flexContainer"
                      role="tablist"
                    >
                      <button
                        aria-controls="tabpanel-one"
                        aria-selected="true"
                        class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
                        data-gtm="{\\"id\\":\\"selector-tabs1\\",\\"label\\":\\"MC: sdp.leadBlock.about\\",\\"action\\":\\"Selector – Tabs\\"}"
                        id="tab-one"
                        role="tab"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiTab-wrapper"
                        >
                          MC: sdp.leadBlock.about
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <span
                      class="PrivateTabIndicator-root-1 PrivateTabIndicator-colorPrimary-2 MuiTabs-indicator"
                      style="left: 0px; width: 0px;"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="tab-one"
            class="MuiGrid-root TabPanel MuiGrid-item"
            heading="MC: sdp.leadBlock.about"
            id="tabpanel-one"
            role="tabpanel"
          >
            <div
              class="MuiBox-root MuiBox-root-5 TabPanelBox"
            >
              <div
                class="MuiGrid-root aboutLeadBlock LeadBlock MuiGrid-container MuiGrid-spacing-xs-3"
              >
                <div
                  class="MuiGrid-root LeadBlockContent MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-8"
                >
                  <div
                    class="LeadBlockContentSection"
                  >
                    <div
                      class="MuiTypography-root Typography MuiTypography-body1"
                    >
                      <p>
                        Long Dec System 1 NO
                      </p>
                    </div>
                  </div>
                  <div
                    class="guaranteesAndAwardsAsset LeadBlockContentSection"
                  >
                    <h5
                      class="MuiTypography-root LeadBlockContentHeading Typography MuiTypography-h5"
                    >
                      MC: sdp.leadBlock.guaranteesWarranties
                    </h5>
                    <img
                      alt="Garanti Test"
                      class="image"
                      src="https://bmipimngqa.azureedge.net/sys-master-hybris-media/hc2/h3e/8972787286046/Monier-garanti-30-aarjpg"
                    />
                  </div>
                  <div
                    class="guaranteesAndAwardsAsset LeadBlockContentSection"
                  >
                    <h5
                      class="MuiTypography-root LeadBlockContentHeading Typography MuiTypography-h5"
                    >
                      MC: sdp.leadBlock.awardsCertificates
                    </h5>
                    <img
                      alt="Award_Test"
                      class="image"
                      src="https://bmipimngqa.azureedge.net/sys-master-hybris-media/h88/h6b/9020240789534/Test-Awardpng"
                    />
                  </div>
                </div>
                <div
                  class="MuiGrid-root item MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-4"
                >
                  <div
                    class="ColorPair ColorPair--pearl"
                  >
                    <div
                      class="PostItCard"
                    >
                      <div
                        class="section"
                      >
                        <h4
                          class="MuiTypography-root Typography MuiTypography-h4"
                        >
                          <h4
                            class="MuiTypography-root Typography Typography--underline MuiTypography-h4"
                          >
                            Key Features
                          </h4>
                        </h4>
                        <div
                          class="content"
                        >
                          <ul
                            class="IconList"
                          >
                            <li
                              class="ListItem ListItem--isCompact"
                            >
                              <div
                                class="icon"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root blueCheckIcon"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                                  />
                                </svg>
                              </div>
                              <div
                                class="content"
                              >
                                <p
                                  class="MuiTypography-root title Typography MuiTypography-body1"
                                >
                                  Sample KF value
                                </p>
                              </div>
                            </li>
                            <li
                              class="ListItem ListItem--isCompact"
                            >
                              <div
                                class="icon"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root blueCheckIcon"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                                  />
                                </svg>
                              </div>
                              <div
                                class="content"
                              >
                                <p
                                  class="MuiTypography-root title Typography MuiTypography-body1"
                                >
                                  Sample KF value 2
                                </p>
                              </div>
                            </li>
                          </ul>
                        </div>
                      </div>
                      <div
                        class="section"
                      >
                        <h4
                          class="MuiTypography-root Typography MuiTypography-h4"
                        >
                          <h4
                            class="MuiTypography-root Typography Typography--underline MuiTypography-h4"
                          >
                            MC: sdp.leadBlock.systemBenefits
                          </h4>
                        </h4>
                        <div
                          class="content"
                        >
                          <ul
                            class="IconList"
                          >
                            <li
                              class="ListItem ListItem--isCompact"
                            >
                              <div
                                class="icon"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root blueCheckIcon"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                                  />
                                </svg>
                              </div>
                              <div
                                class="content"
                              >
                                <p
                                  class="MuiTypography-root title Typography MuiTypography-body1"
                                >
                                  Benefit value 1
                                </p>
                              </div>
                            </li>
                            <li
                              class="ListItem ListItem--isCompact"
                            >
                              <div
                                class="icon"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root blueCheckIcon"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                                  />
                                </svg>
                              </div>
                              <div
                                class="content"
                              >
                                <p
                                  class="MuiTypography-root title Typography MuiTypography-body1"
                                >
                                  Benefit value 2
                                </p>
                              </div>
                            </li>
                          </ul>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TabLeadBlock tests should render the bimIframe tab 1`] = `
<div
  aria-labelledby="tab-four"
  class="MuiGrid-root TabPanel MuiGrid-item"
  heading="MC: sdp.tabs.bim"
  hidden=""
  id="tabpanel-four"
  role="tabpanel"
>
  <div
    class="MuiBox-root MuiBox-root-11 TabPanelBox"
  >
    <iframe
      class="bimIframe"
      src="https://google.com"
    />
  </div>
</div>
`;
